### ISSUE TITLE
How to further fine-tune on custom data ?

### ISSUE THREAD

**Issue by rakshit-ce** [hpanwar08/detectron2#31]
I've gone through `https://github.com/hpanwar08/detectron2/blob/master/tools/train_net_dla.py` training script. But, I'm bit confused on how to structure my custom data (both JSON and IMG-data).

If you can share a sample JSON and folder structure for placing training images, it would be great !!

**Comment by hpanwar08** [comment]
You can use the same `train_net_dla.py` to finetune if your data is in COCO format. You can convert your data to coco easily.

COCO format has an annotation json file and a folder that contains all the images. Read more about it [here]( https://www.immersivelimit.com/tutorials/create-coco-annotations-from-scratch)

If you do not wish to convert to coco, then go through the colab notebook from original detectron2 for training on custom data format.

**Comment by rakshit-ce** [comment]
I don't have the segmentation data, so will populating only `bbox` field in json be enough ??

**Comment by hpanwar08** [comment]
Yes, bounding box is enough. Use Faster RCNN for bounding box only data.

**Comment by rakshit-ce** [comment]
There are no pre-trained weights available in this repo for Faster RCNN, does this mean I'll have to train from scratch on `publaynet` dataset and then fine-tune on my custom dataset ?

I'm asking because I wanted to fine-tune your publaynet-trained models further on my data !!

**Comment by hpanwar08** [comment]
One work around you can do it to use Mask RCNN pretrained weights from this repo and in your dataset use the bounding box as mask.

**Comment by rakshit-ce** [comment]
How can I convert bounding box to mask ?

**Comment by hpanwar08** [comment]
There is no explicit conversion required, you can use bounding box as mask. If you are using coco format then copy the bounding box values in the segmentation part as an array.
